app/uk/gov/hmrc/agentsubscriptionfrontend/service/SubscriptionService.scala

```diff
@@ -61,4 +82,50 @@ class SubscriptionService @Inject()(agentSubscriptionConnector: AgentSubscriptio
       case e => throw e
     }
   }
+
+  def completePartialSubscription(details: CompletePartialSubscriptionBody)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[(Boolean, Option[Arn])] =
+    for {
+      subscriptionProgress <- getSubscriptionStatus(details.utr, details.knownFacts.postcode) //FixMe APB-2805 this check should happen previously and can be stored in the session
+      isPartiallySubscribed = subscriptionProgress.state == SubscriptionState.IsOnlySubscribedInETMP
+      (wasPartiallySubscribed, optSucceededAllocatingArn) <- isPartiallySubscribed match {
+                                                              case true =>
```

Comment:
Pattern matching on the boolean `isPartiallySubscribed` but could use `if(isPartiallySubscribed) {} else {}`